common/prque/sstack.go

```diff
@@ -94,7 +103,35 @@ func (s *sstack) Len() int {
 // Compares the priority of two elements of the stack (higher is first).
 // Required by sort.Interface.
 func (s *sstack) Less(i, j int) bool {
-	return (s.blocks[i/blockSize][i%blockSize].priority - s.blocks[j/blockSize][j%blockSize].priority) > 0
+	iPriority := s.blocks[i/blockSize][i%blockSize].priority
+	jPriority := s.blocks[j/blockSize][j%blockSize].priority
+
+	return less(iPriority, jPriority, s.invert)
+}
+
+func less(i, j interface{}, invert bool) bool {
+	var result bool
+	switch iPriority := i.(type) {
+	case int:
+		// If an type assertion error occurred, check types in Push().
+		// Same type should be pushed.
+		jPriority := j.(int)
+		result = iPriority > jPriority
+	case int64:
+		jPriority := j.(int64)
+		result = iPriority > jPriority
+	case uint64:
+		jPriority := j.(uint64)
+		result = iPriority > jPriority
+	case []byte:
+		jPriority := j.([]byte)
+		result = bytes.Compare(iPriority, jPriority) > 0
+	}
+
+	if invert {
+		return !result
+	}
+	return result
```

Comment:
Might be harder on the eyes, but you could just xor `result` with `invert` and return that, sth like `return invert ^ result`. However, there's no xor operator for booleans, so it would instead be something lke `return result != invert`